bootstrap.js

```diff
@@ -4,101 +4,206 @@ Cu.import("resource://gre/modules/Console.jsm");
 Cu.import("resource://gre/modules/AppConstants.jsm");
 
 const CSS_URI = Services.io.newURI("resource://share-button-study/share_button.css");
-let bWindow = null;
-let bDocument = null;
-let utils = null;
-let shareButton = null;
-let urlInputControllers = null;
-
-function shareButtonAnimationListener(e) {
-  // When the animation is done, we want to remove the CSS class
-  // so that we can add the class again upon the next copy and
-  // replay the animation
-  shareButton.classList.remove("social-share-button-on");
+const browserWindowWeakMap = new WeakMap();
+
+function getUrlInput(bWindow) {
+  // Get the "DOM" elements
+  const urlBar = bWindow.document.getElementById("urlbar");
+  // XUL elements are different than regular children
+  const urlInput = bWindow.document.getAnonymousElementByAttribute(urlBar, "anonid", "input");
+  return urlInput;
+}
+
+class BrowserWindow {
+  constructor(aWindow) {
+    this.window = aWindow;
+    this.urlInput = getUrlInput(this.window);
+    this.shareButton = this.window.document.getElementById("social-share-button");
+
+    // bind functions that are called externally so that `this` will work
+    this.shareButtonListener = this.shareButtonListener.bind(this);
+
+    this.insertCopyController = this.insertCopyController.bind(this);
+    this.removeCopyController = this.removeCopyController.bind(this);
+
```

Comment:
I had them grouped by `shareButton`, `copyController`, and `customizeListener`, but that was mostly for myself during development.